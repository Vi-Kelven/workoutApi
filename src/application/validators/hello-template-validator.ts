import { joiBase } from './base/joi-base';

export const helloTemplateSchema = joiBase.joi.object(
    {
        nome: joiBase.joi.string().required(),
    }
)
