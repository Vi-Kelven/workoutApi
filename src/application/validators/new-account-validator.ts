import { rolesEnum } from '../../domain/entity/enum/roles-enum';
import { joiBase } from './base/joi-base';

export const newAccountValidator = joiBase.joi.object(
    {
        nome: joiBase.joi.string().required(),
        username: joiBase.joi.string().required(),
        email: joiBase.joi.string().email({tlds: {allow: false}}).required(),
        senha: joiBase.joi.string().required(),
        tipo_usuario: joiBase.joi.string().valid(...Object.values(rolesEnum)).required()
    }
)

export const updateAccountValidator = joiBase.joi.object(
    {
        id: joiBase.joi.string().required(),
        nome: joiBase.joi.string(),
        username: joiBase.joi.string(),
        email: joiBase.joi.string().email({tlds: {allow: false}}),
        senha: joiBase.joi.string(),
        tipo_usuario: joiBase.joi.string().valid(...Object.values(rolesEnum))
    }
)

export const deleteAccountValidator = joiBase.joi.object(
    {
        id: joiBase.joi.string().required(),
        nome: joiBase.joi.string(),
        username: joiBase.joi.string(),
        email: joiBase.joi.string().email({tlds: {allow: false}}),
        senha: joiBase.joi.string(),
        tipo_usuario: joiBase.joi.string().valid(...Object.values(rolesEnum))
    }
)

