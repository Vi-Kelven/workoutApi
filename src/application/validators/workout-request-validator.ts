import { joiBase } from "./base/joi-base";

export const WorkoutRequestValidator = joiBase.joi.object({
    user: joiBase.joi.object({
        nome: joiBase.joi.string(),
        idade: joiBase.joi.number(),
        sexo: joiBase.joi.string(),
        objetivo: joiBase.joi.string(),
        nivel: joiBase.joi.string(),
        experiente: joiBase.joi.boolean(),
        localidade: joiBase.joi.string(),
        equipamentos: joiBase.joi.string(),
        tempo: joiBase.joi.number(),
        frequencia: joiBase.joi.number(),
        restrição: joiBase.joi.string(),
        duracaoTreino: joiBase.joi.string(),
        cardio: joiBase.joi.boolean(),
        funcional: joiBase.joi.boolean()
    }).required(),
    muscle: joiBase.joi.array().items(
        joiBase.joi.string()
    ).min(1)
}).required()
