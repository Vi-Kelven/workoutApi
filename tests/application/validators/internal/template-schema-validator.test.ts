import { helloTemplateSchema } from '../../../../src/application/validators/hello-template-validator';

describe('template schema', () => {
  test('deve validar os campos incorretos', () => {
    const response = helloTemplateSchema.validate(
      { nome: true }, 
      { abortEarly: false }
    );
    expect(response?.error?.details[0].message).toBe('"nome" deve ser um texto')
    expect(response?.error?.details).toMatchSnapshot();
  });

  test('deve validar os campos obrigatórios', () => {
    const response = helloTemplateSchema.validate({}, { abortEarly: false });
    expect(response?.error?.details[0].message).toBe(`"nome" é um campo obrigatório`)
    expect(response?.error?.details).toMatchSnapshot();
  });

  test('deve validar um caso de sucesso', () => {
    const response = helloTemplateSchema.validate(
      { nome: "Ciclano" }, 
      { abortEarly: false }
    );
    expect(response?.value).toMatchSnapshot();
  });
});
