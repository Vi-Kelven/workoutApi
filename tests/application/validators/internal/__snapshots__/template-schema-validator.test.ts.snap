// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template schema deve validar os campos incorretos 1`] = `
[
  {
    "context": {
      "key": "nome",
      "label": "nome",
      "value": true,
    },
    "message": ""nome" deve ser um texto",
    "path": [
      "nome",
    ],
    "type": "string.base",
  },
]
`;

exports[`template schema deve validar os campos obrigatórios 1`] = `
[
  {
    "context": {
      "key": "nome",
      "label": "nome",
    },
    "message": ""nome" é um campo obrigatório",
    "path": [
      "nome",
    ],
    "type": "any.required",
  },
]
`;

exports[`template schema deve validar um caso de sucesso 1`] = `
{
  "nome": "Ciclano",
}
`;
